### PR TITLE
Ibce preview

### PR DESCRIPTION
**Pull request by DavieReid**
Refactors required for In browser content editor to become functional.

Another key change is that `mosaic.config.js` has been renamed everywhere to `mosaic.config.mjs`.  This allows us to import workflows into the config file but also is consistent with our approach of trying to have "ESM everywhere".

Hi @origami-z

### CODE REVIEW

File: packages/site/mosaic.config.mjs [hunk 1]
```diff
@@ -53,6 +45,20 @@ module.exports = deepmerge(siteConfig, {
         prefixDir: 'mosaic', // root path used for namespace
         extensions: ['.mdx'] // extensions of content which should be pulled
       }
+    },
```
Comment (DavieReid) [inline packages/site/mosaic.config.mjs:48]: oops will remove

File: packages/site/src/pages/api/content/preview.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { compileMDX } from '@jpmorganchase/mosaic-site-middleware';
```
Comment (DavieReid) [inline packages/site/src/pages/api/content/preview.ts:1]: Not sure if this should be removed (as we don't need it) or if it should be kept and also added to the site generator template...

This the endpoint the preview component hits to compile the markdown pulled from the IBCE

thoughts? @lukeac123 @lilyvc @mark-tate @tomhazledine

File: packages/site/mosaic.config.mjs (hunk 1 above)
Comment (DavieReid) [reply packages/site/mosaic.config.mjs]: DONE

File: packages/site/.env.local [hunk 3]
```diff
@@ -3,4 +3,7 @@ MOSAIC_SNAPSHOT_DIR=snapshots/latest
 OPTIMIZE_IMAGES=false
 NEXTAUTH_URL=http://localhost:3000
 NODE_ENV=development
-MOSAIC_ENABLE_SOURCE_PUSH=true
\ No newline at end of file
+MOSAIC_ENABLE_SOURCE_PUSH=true
+NEXTAUTH_URL=http://localhost:3000/
```
Comment (libertymayc) [inline packages/site/.env.local:7]: repeat of above? or you need one with a slash?

File: packages/site/src/pages/api/content/preview.ts (hunk 2 above)
Comment (lukeac123) [reply packages/site/src/pages/api/content/preview.ts:1]: If it's not being used what's the reasoning to keep it ?

File: packages/site/.env.local (hunk 3 above)
Comment (DavieReid) [reply packages/site/.env.local:7]: no copy & paste error...will remove in next PR

Comment (lukeac123) [reply packages/site/.env.local:7]: think it works with either

**Review by lukeac123 — APPROVED** [review]

**Review by lilyvc — APPROVED** [review]

**Pull request merged**